bftengine/src/preprocessor/messages/PreProcessResultMsg.cpp

```diff
@@ -70,7 +70,15 @@ std::optional<std::string> PreProcessResultMsg::validatePreProcessResultSignatur
   }
 
   auto hash = concord::util::SHA3_256().digest(requestBuf(), requestLength());
+  std::unordered_set<bftEngine::impl::NodeIdType> seen_signatures;
   for (const auto& s : sigs) {
+    const auto [it, created] = seen_signatures.insert(s.sender_replica);
```

Comment:
Very minor comment.
Instead of keeping an unused variable can we do
if(seen_signature.find(s.sender_replica) != seen_signature.end()) {
   // its present logic
} else {
   seen_signatures.insert(s.sender_replica);
}
This way we will avoid the unused variable.

Or we can also do the following to avoid the find:
if(!seen_signatures.insert(s.sender_replica).second) {
     // do the error logic. 
}

Reply:
I wanted to avoid searching and then inserting in the set, but your second idea is very good. I'll do it that way.